node/daser/module.go

```diff
@@ -0,0 +1,21 @@
+package daser
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/node/node"
+)
+
+func Module(tp node.Type) fx.Option {
+	switch tp {
+	case node.Light, node.Full:
+		return fx.Module(
+			"daser",
+			fx.Provide(DASer),
+		)
+	case node.Bridge:
+		return fx.Module("daser")
```

Comment:
this might be annoying for when we provide the top-level API via rpc. But we will see.

Reply:
Instead, I would return `fx.Options()` there